[quant] Add QuantFusion to graph_executor

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26591 [quant] Add QuantFusion to graph_executor**

Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17636651](https://our.internmc.facebook.com/intern/diff/D17636651)

jerryzh168 committed:
[quant] Add QuantFusion to graph_executor

Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
btw this PR is regenerated, last PR is merged somehow.

ZolotukhinM commented on torch/csrc/jit/graph_executor.cpp:
```diff
@@ -708,6 +711,11 @@ void runNondiffOptimization(std::shared_ptr<Graph>& graph) {
   BatchMM(graph);
 
+  if (at::globalContext().qEngine() == at::kFBGEMM ||
+      at::globalContext().qEngine() == at::kQNNPACK) {
```
Optimization pipeline in graph executor cannot be conditionalized on a global state. We need to make sure QuantFusion is always correct and run it unconditionally.

jerryzh168 committed:
Update on "[quant] Add QuantFusion to graph_executor"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/graph_executor.cpp:
```diff
@@ -1,5 +1,7 @@
 #include <torch/csrc/jit/graph_executor.h>
 
+#include <aten/src/ATen/Context.h>
+
```
This is unused.

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant] Add QuantFusion to graph_executor"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
